minsar/utils/download_asfserial_rsmas.py

```diff
@@ -67,15 +88,64 @@ def main(iargs=None):
+        y = tupleParam[4]
+        YearRange = tupleParam[5]
+        seasonalStartDateAddOn = tupleParam[6]
+        seasonalEndDateAddOn = tupleParam[7]
+        ogEndDate = tupleParam[8]
+        if x == 0:
+            if YearRange == 1:
+                if y == 0:
+                    if int(inps.seasonalEndDate) < int(dataset_template.options['ssaraopt.startDate'][4:].replace('-', '')) or int(inps.seasonalStartDate) > int(ogEndDate[4:].replace('-', '')):
+                        return
+                    else:
+                        if int(inps.seasonalStartDate) >  int(dataset_template.options['ssaraopt.startDate'][4:].replace('-', '')):
+                            dataset_template.options['ssaraopt.startDate'] = str(ogStartYearInt) + seasonalStartDateAddOn
+                        if int(inps.seasonalEndDate) < int(ogEndDate[4:].replace('-', '')):
+                            dataset_template.options['ssaraopt.endDate'] = str(ogStartYearInt) + seasonalEndDateAddOn
+                elif int(dataset_template.options['ssaraopt.endDate'][:4]) - ogStartYearInt + 1 == 1:
+                    if int(inps.seasonalStartDate) > int(ogEndDate[4:].replace('-', '')):
+                        return
+                    elif int(inps.seasonalStartDate) >  int(dataset_template.options['ssaraopt.startDate'][4:].replace('-', '')):
+                        dataset_template.options['ssaraopt.startDate'] = str(ogStartYearInt) + seasonalStartDateAddOn
+                else:
+                    if int(inps.seasonalStartDate) >  int(dataset_template.options['ssaraopt.startDate'][4:].replace('-', '')):
+                        dataset_template.options['ssaraopt.startDate'] = str(ogStartYearInt) + seasonalStartDateAddOn
+                    if int(inps.seasonalEndDate) < int(ogEndDate[4:].replace('-', '')):
+                        dataset_template.options['ssaraopt.endDate'] = str(ogStartYearInt + y) + seasonalEndDateAddOn
+            else:
+                if y == 0:
+                    if int(inps.seasonalEndDate) < int(dataset_template.options['ssaraopt.startDate'][4:].replace('-', '')):
+                        return
+                    else: 
+                        if int(inps.seasonalStartDate) >  int(dataset_template.options['ssaraopt.startDate'][4:].replace('-', '')):
+                            dataset_template.options['ssaraopt.startDate'] = str(ogStartYearInt) + seasonalStartDateAddOn
+                        dataset_template.options['ssaraopt.endDate'] = str(ogStartYearInt) + seasonalEndDateAddOn
+                else:
+                    if int(inps.seasonalStartDate) >  int(dataset_template.options['ssaraopt.startDate'][4:].replace('-', '')):
+                        dataset_template.options['ssaraopt.startDate'] = str(ogStartYearInt) + seasonalStartDateAddOn
+                    dataset_template.options['ssaraopt.endDate'] = str(ogStartYearInt + y) + seasonalEndDateAddOn
+        elif x < YearRange - 1:
```

Comment:
Why the `-1`? Could you manipulate x off the bat to be 1 less instead of having this strange looking condition?

Reply:
The `-1` is there because of the nature of the range function in python - range(2) 0 -> 1. The number of loops (equal to years in one of the two cases) is one less than the YearRange. x has a meaningful value (the loop counter + year increment to the original start date). YearRange has a less meaningful value. I could reduce x but that would mean a `+1` at each date mutation. By reducing YearRange, a `+1` will be needed at the range function.